src/java/com/puppetlabs/jruby_utils/pool/JRubyPool.java

```diff
@@ -293,8 +331,12 @@ public void lock() throws InterruptedException {
                 }
             }
             try {
-                while (registeredElements.size() != liveQueue.size()) {
-                    allRegisteredInQueue.await();
+                //
```

Comment:
Nit, seems like this empty comment could go away - or you could replace it with something which says we want to hold here until the pool has been filled to capacity.